src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java

```diff
@@ -492,7 +471,17 @@ private final class InternalLoader
          */
         private InternalLoader()
                 throws SAXException, ParserConfigurationException {
-            super(createIdToResourceNameMap());
+            // map from local resources to dtd ids
+            super(Map.of(
```

Comment:
this is hacking to only cheat Pitest.

but conceptually this collection is required to load old config files that have reference to old schema name.

can we create config xml file with `-//Puppy Crawl//DTD Check Configuration 1.0//EN` instead of `-//Checkstyle//DTD Checkstyle Configuration 1.0//EN` and make sure config is loadable.

This test is ok to be not Input based.

Reply:
> this is hacking to only cheat Pitest.

I disagree. It looks like the method that built this map was written in 2015, before we migrated to JDK11. `Map.of` was released in JDK9. The method we removed did nothing but create a map anyway; we are doing better now that we are returning an Unmodifiable map. We can write any tests we want, but the fact is that passing an Unmodifiable map to the super constructor is better that what we were doing previously.


Since we can have no code before the super constructor call, pre-JDK9 code required some other method to exist. We do not need it any longer.

Reply:
I have no problem to make `createIdToResourceNameMap()` to return immutable map.

Pitest clearly point us that we have no test for old configs.

Reply:
@Kevin222004 , please finish this PR.

Reply:
Done Test added